langchain-core/src/runnables/base.ts

```diff
@@ -1663,9 +1672,9 @@ export function _coerceToRunnable<RunInput, RunOutput>(
   } else if (!Array.isArray(coerceable) && typeof coerceable === "object") {
     const runnables: Record<string, Runnable<RunInput>> = {};
     for (const [key, value] of Object.entries(coerceable)) {
-      runnables[key] = _coerceToRunnable(value);
+      runnables[key] = _coerceToRunnable(value as RunnableLike);
     }
-    return new RunnableMap<RunInput>({
+    return new RunnableMap({
```

Comment:
Why did you drop the generic passing here?

Reply:
So when passing as RunnableMap<RunInput, RunOutput>, the type of Run output does not line up until the line below
```
    }) as unknown as Runnable<RunInput, Exclude<RunOutput, Error>>;
```

Passing _just_ the RunInput defaults the RunOutput to Record<string, any>. So you end up having to pass both values (which is not possible to the reason above) or neither